src/main/java/codecain/GraphicalUserInterface/View/MenuGUI.java

```diff
@@ -10,58 +10,97 @@
-                cliStage.setOnCloseRequest(event1 -> primaryStage.show());
-            } catch (Exception e) {
-                e.printStackTrace();
-            }
+            launchCLI(primaryStage); 
         });
 
-        // GUI Button
         Button guiButton = new Button("Graphical Interface");
         guiButton.setOnAction(event -> {
-            try {
-                Stage guiStage = new Stage();
-                GraphicalInterfaceJavaFX gui = new GraphicalInterfaceJavaFX();
-                gui.start(guiStage); // Launch GUI directly
-                primaryStage.hide(); // Hide the main menu
-
-                // Show the main menu again when the GUI stage is closed
-                guiStage.setOnCloseRequest(event1 -> primaryStage.show());
-            } catch (Exception e) {
-                e.printStackTrace();
-            }
+            launchGraphicalInterface(primaryStage);
         });
-
-        // Layout setup
         VBox layout = new VBox(20, welcomeText, cliButton, guiButton);
-        layout.setAlignment(Pos.CENTER); // Center everything horizontally and vertically
-
-        // Scene setup
+        layout.setAlignment(Pos.CENTER);
         Scene scene = new Scene(layout, 400, 300);
         primaryStage.setTitle("UML Manager");
         primaryStage.setScene(scene);
         primaryStage.show();
     }
 
+    /**
+     * Launches the Command Line Interface (CLI) in a new stage.
+     * The main menu is hidden while the CLI window is active and shown again
+     * when the CLI window is closed.
+     *
+     * @param parentStage The main menu stage to hide while the CLI is active.
+     */
+    public static void launchCLI(Stage parentStage) {
+        try {
+            Stage cliStage = new Stage();
+            CLI cli = new CLI();
+            cli.start(cliStage);
+            if (parentStage != null) parentStage.hide(); // Hide the main menu
+
+            cliStage.setOnCloseRequest(event1 -> {
+                if (parentStage != null) parentStage.show();
+            });
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Launches the Graphical User Interface (GUI) in a new stage.
+     * The main menu is hidden while the GUI window is active and shown again
+     * when the GUI window is closed.
+     *
+     * @param parentStage The main menu stage to hide while the GUI is active.
+     */
+    public static void launchGraphicalInterface(Stage parentStage) {
+        try {
+            Stage guiStage = new Stage();
+            GraphicalInterfaceJavaFX gui = new GraphicalInterfaceJavaFX();
+            gui.start(guiStage);
+            if (parentStage != null) parentStage.hide();
+
+            guiStage.setOnCloseRequest(event1 -> {
+                if (parentStage != null) parentStage.show();
+            });
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
```

Comment:
Did you create this to use the GUIs export function then hide the GUI so its the CLI command that is making the call to it?

Reply:
Yes that was the focus of including this rework but it also allows us to open up the GUI and CLI for other future features such as the Duel GUI and CLI mode we were talking about.